provider/datagen/src/baked_exporter.rs

```diff
@@ -332,80 +334,98 @@ impl DataExporter for BakedExporter {
             }
-            _ => {
-                let mut map = BTreeMap::new();
-                let mut statics = Vec::new();
-
-                for (bake, locales) in values {
-                    let first_locale = locales.iter().next().unwrap();
-                    let anchor = syn::parse_str::<syn::Ident>(
-                        &first_locale.to_ascii_uppercase().replace('-', "_"),
-                    )
-                    .unwrap();
-                    statics.push(quote! { static #anchor: #struct_type = #bake; });
-                    map.extend(locales.into_iter().map(|l| (l, anchor.clone())));
-                }
+        } else {
+            let mut map = BTreeMap::new();
+            let mut statics = Vec::new();
+
+            for (bake, locales) in values {
+                let first_locale = locales.iter().next().unwrap();
+                let anchor = syn::parse_str::<syn::Ident>(
+                    &first_locale.to_ascii_uppercase().replace('-', "_"),
+                )
+                .unwrap();
+                statics.push(quote! { static #anchor: #struct_type = #bake; });
+                map.extend(locales.into_iter().map(|l| (l, anchor.clone())));
+            }
+
+            let (keys, values): (Vec<_>, Vec<_>) = map.into_iter().unzip();
 
-                let (keys, values): (Vec<_>, Vec<_>) = map.into_iter().unzip();
+            let lookup = quote! {
+                #(#statics)*
+                match [#(#keys),*].binary_search_by(|k| locale.strict_cmp(k.as_bytes()).reverse()) {
+                    Ok(i) => Ok(*unsafe {
+                        // Safe because keys and values have the same length
+                        [#(&#values),*].get_unchecked(i)
+                    }),
+                    Err(_) => Err(icu_provider::DataErrorKind::MissingLocale)
+                }
+            };
 
-                quote! {
-                    #(#statics)*
-                    match [#(#keys),*].binary_search_by(|k| locale.strict_cmp(k.as_bytes()).reverse()) {
-                        Ok(i) => Ok(*unsafe {
-                            // Safe because keys and values have the same length
-                            [#(&#values),*].get_unchecked(i)
-                        }),
-                        Err(_) => Err(icu_provider::DataErrorKind::MissingLocale)
+            match fallback_mode {
+                FallbackMode::None => lookup,
+                FallbackMode::Full => {
+                    quote! {
+                        const FALLBACKER: icu_provider_adapters::fallback::LocaleFallbackerWithConfig = icu_provider_adapters::fallback::LocaleFallbackerBorrowed::from_parts(
+                            <$provider>::SINGLETON_FALLBACK_LIKELYSUBTAGS_V1,
+                            <$provider>::SINGLETON_FALLBACK_PARENTS_V1,
+                            Some(<$provider>::SINGLETON_FALLBACK_SUPPLEMENT_CO_V1)
+                        ).for_key(<#marker as icu_provider::KeyedDataMarker>::KEY);
```

Comment:
Praise: This is better than calling the macros (if we move forward with the approach of constructing LocaleFallbackers for each key). We should save on compile times, but I still worry whether the `const` will be inlined everywhere (since it's not a `static`).

Reply:
The const is a reference to an anonymous static (`&'static Foo`). I believe this gives the compiler more freedom than if it were a static.

Reply:
Yes, but where exactly is the data being anchored in memory? I suspect that we could still end up with a case where it gets anchored as a local `static` into each call site of the `const`.

Perhaps @Manishearth can verify how this happens.

Reply:
It's hard to verify for sure without looking at the code because LLVM optimizes in most cases. Looking at the MIR I _think_ the const-promoted static gets evaluated first so we should be fine.